cpp/src/arrow/python/python_to_arrow.cc

```diff
@@ -1004,6 +991,9 @@ Result<std::shared_ptr<ChunkedArray>> ConvertPySequence(PyObject* obj, PyObject*
   PyObject* seq;
   OwnedRef tmp_seq_nanny;
 
+  // FIXME(kszucs): shouldn't import pandas unconditionally
```

Comment:
We import pandas when converting from arrow to python in case of nanosecond resolution timestamp so to have successful roundtrip we need pandas to convert it back.

Reply:
We could make it depend on `options.from_pandas`, but currently we use that flag for null checks only.

Reply:
@jorisvandenbossche any suggestions here?

Reply:
I think we can simply remove this initialization since pandas should be already imported to have pandas objects in the input data.

Reply:
Perhaps only initialize pandas static data if pandas has been already imported (in sys.modules)?

Reply:
Updated according to my previous comment.